docs/src/architecture/09_architecture_decisions/0006-rust-cargo-lock.md

```diff
@@ -0,0 +1,45 @@
+---
+    title: 0006 Rust Cargo Lock
+    adr:
+        author: Steven Johnson
+        created: 09-Jan-2024
+        status:  accepted
+        extends:
+            - 0005-rust
+    tags:
+        - Rust
+---
+
+## Context
+
+Rust has an optional `cargo.lock` file which can lock the dependencies of a project.
+There are [pros and cons][Cargo/toml vs Cargo.lock] to using the `lock` file.
+
+## Assumptions
+
+This ADR is deliberately limited to the initial bring up phase of our projects, and subject to review.
+
+## Decision
+
+Rust will not use `cargo.lock` when building libraries.  
+It will ONLY respect it for building binaries.
+
+As we are in the initial stages of a number of projects, it is easier to iterate without
+worrying about `cargo.lock` being up-to-date.
```

Comment:
Doesn't this make collaboration harder? IIRC, it's "locked" at the time of running Cargo, meaning devs could be using a number of different dependencies.

Reply:
It came up because it was checked in haphazardly and was causing CI failures.  It seems an unnecessary headache for now given how early we are, the fact it won't control dependencies on libraries and we are still rapidly iterating on code.

Reply:
The dependencies are described by `cargo.toml` the lock is "at the time of build" only.
So the dependencies shouldn't change.
It can also cause issues when versions are yanked upstream.